_posts/2022/2022-11-14-safely-remove-a-column-field-from-active-record.md

```diff
@@ -0,0 +1,71 @@
+---
+title: "Safely Remove a Column from an Active Record Model"
+description: "Avoid migration-related errors when you’re tidying your database"
+layout: article
+category: ruby
+image:
+  base: "2022/safely-remove-a-column-field-from-active-record"
+  alt: "A partial column against flat background"
+  credit: "ANIRUDH"
+  source: "https://unsplash.com/photos/jtiQCAAzLYY"
+---
+
+Adding and deploying new columns to an existing Active Record model is often straightforward. Typically the first deployment runs your migrations and then new code that uses the new database column is released afterward.
+
+However, when removing a column you can more easily generate problems. Active Record caches database columns when it spins up a Rails application. When the column is removed from the database table it causes exceptions until your app reboots or redeploys.
+
+It's worth a multi-step strategy when removing columns in your database.
+
+
+## Use...
+
+...this multi-step strategy.
+
+First tell Active Record to ignore the column. This stops the rest of your application being able to reference this column.
```

Comment:
Isn't the first step really removing all code that references the old column?

Reply:
No, you can find the places you use it by using the test suite!

Reply:
Oh, yeah, I see what you mean!